ibis/expr/types/json.py

```diff
@@ -98,6 +100,132 @@ def __getitem__(
+        ... }
+        >>> t = ibis.memtable(data, schema=dict.fromkeys(data.keys(), "json"))
+        >>> t
+        ┏━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┳━━━┓
+        ┃ jstring              ┃ jbool                ┃ jint                 ┃ … ┃
+        ┡━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━╇━━━┩
+        │ json                 │ json                 │ json                 │ … │
+        ├──────────────────────┼──────────────────────┼──────────────────────┼───┤
+        │ 'a'                  │ True                 │ 1                    │ … │
+        │ ''                   │ False                │ None                 │ … │
+        │ NULL                 │ None                 │ NULL                 │ … │
+        │ None                 │ NULL                 │ 2                    │ … │
+        └──────────────────────┴──────────────────────┴──────────────────────┴───┘
+        >>> t.select(unwrapped=t.jstring.unwrap_as(str), original=t.jstring)
+        ┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ unwrapped ┃ original             ┃
+        ┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━┩
+        │ string    │ json                 │
+        ├───────────┼──────────────────────┤
+        │ a         │ 'a'                  │
+        │ ~         │ ''                   │
+        │ NULL      │ NULL                 │
+        │ NULL      │ None                 │
+        └───────────┴──────────────────────┘
+        >>> t.select(unwrapped=t.jbool.unwrap_as("bool"), original=t.jbool)
+        ┏━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ unwrapped ┃ original             ┃
+        ┡━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━┩
+        │ boolean   │ json                 │
+        ├───────────┼──────────────────────┤
+        │ True      │ True                 │
+        │ False     │ False                │
+        │ NULL      │ None                 │
+        │ NULL      │ NULL                 │
+        └───────────┴──────────────────────┘
+        >>> t.select(
+        ...     unwrapped_int64=t.jint.unwrap_as("int64"),
+        ...     unwrapped_int32=t.jint.unwrap_as("int32"),
+        ...     original=t.jint,
+        ... )
+        ┏━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┓
+        ┃ unwrapped_int64 ┃ unwrapped_int32 ┃ original             ┃
+        ┡━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━┩
+        │ int64           │ int32           │ json                 │
+        ├─────────────────┼─────────────────┼──────────────────────┤
+        │               1 │               1 │ 1                    │
+        │            NULL │            NULL │ None                 │
+        │            NULL │            NULL │ NULL                 │
+        │               2 │               2 │ 2                    │
+        └─────────────────┴─────────────────┴──────────────────────┘
+        >>> t.select(unwrapped=t.jfloat.unwrap_as("float32"), original=t.jfloat)
```

Comment:
Shouldn't this be``"float64"``? I don't see an unwrap method for ``"float32"``, but the return below is also ``"float32"`` which seems interesting?

Reply:
You can give as specific a type as you want and Ibis will try to cast the result to that type.

The `float32` here is meant to show that you can use a more specific floating point type as well as the more-typical `float`/`float64` (`float` is an alias for `float64`).

I'll add some prose to the docstrings.